src/Microsoft.AspNet.Server.Kestrel/Http/Connection.cs

```diff
@@ -104,9 +146,9 @@ private void OnRead(UvStreamHandle handle, int status, Exception error)
                 }
             }
 
-
             try
             {
+                ExtractIP();
```

Comment:
Huh? KeepAlive means the underlying TCP connection is not closed between two requests. This means both the remote port and the server IP address cannot change, because then this would be a different TCP connection.

KeepAlive as implemented on `dev` is currently broken. Maybe you were chasing those problems by accident?